contracts/contracts/strategies/sonic/SonicValidatorDelegator.sol

```diff
@@ -0,0 +1,354 @@
+     * @param validatorId The Sonic validator ID to undelegate from.
+     * @param undelegateAmount the amount of Sonic (S) to undelegate.
+     */
+    function undelegate(uint256 validatorId, uint256 undelegateAmount)
+        external
+        onlyRegistratorOrStrategist
+        nonReentrant
+        returns (uint256 withdrawId)
+    {
+        // Can still undelegate even if the validator is no longer supported
+        require(undelegateAmount > 0, "Must undelegate something");
+
+        uint256 amountDelegated = sfc.getStake(address(this), validatorId);
+        require(undelegateAmount <= amountDelegated, "Insufficient delegation");
+
+        withdrawId = nextWithdrawId++;
+
+        withdrawals[withdrawId] = WithdrawRequest(
+            validatorId,
+            undelegateAmount,
+            block.timestamp
+        );
+        pendingWithdrawals += undelegateAmount;
+
+        sfc.undelegate(validatorId, withdrawId, undelegateAmount);
+
+        emit Undelegated(withdrawId, validatorId, undelegateAmount);
+    }
+
+    /**
+     * @notice Withdraw native S from a previously undelegated validator.
+     * The native S is wrapped wS and transferred to the Vault.
+     * @param withdrawId The withdraw ID returned from `undelegate` and emitted in `Undelegated`.
+     * @return withdrawnAmount The amount of Sonic (S) withdrawn.
+     * This can be less than the undelegated amount in the event of slashing.
+     */
+    // slither-disable-start reentrancy-no-eth
+    function withdrawFromSFC(uint256 withdrawId)
+        external
+        onlyRegistrator
+        nonReentrant
+        returns (uint256 withdrawnAmount)
+    {
+        require(withdrawId < nextWithdrawId, "Invalid withdrawId");
+
+        // Can still withdraw even if the validator is no longer supported
+        // Load the withdrawal from storage into memory
+        WithdrawRequest memory withdrawal = withdrawals[withdrawId];
+        require(!isWithdrawnFromSFC(withdrawId), "Already withdrawn");
+
+        uint256 sBalanceBefore = address(this).balance;
+
+        sfc.withdraw(withdrawal.validatorId, withdrawId);
+
+        // Save state to storage
+        withdrawnAmount = address(this).balance - sBalanceBefore;
+        pendingWithdrawals -= withdrawal.undelegatedAmount;
+        withdrawals[withdrawId].undelegatedAmount = 0;
+
+        // Wrap Sonic (S) to Wrapped Sonic (wS)
+        IWrappedSonic(wrappedSonic).deposit{ value: withdrawnAmount }();
+
+        // Transfer the Wrapped Sonic (wS) to the Vault
+        // slither-disable-next-line unchecked-transfer unused-return
+        IERC20(wrappedSonic).transfer(vaultAddress, withdrawnAmount);
+
+        // withdrawal.undelegatedAmount & withdrawnAmount can differ in case of slashing
+        emit Withdrawn(
+            withdrawId,
+            withdrawal.validatorId,
+            withdrawal.undelegatedAmount,
+            withdrawnAmount
+        );
+    }
+
+    // slither-disable-end reentrancy-no-eth
```

Comment:
I think slither was picking up on an actual incorrectness in how the function was written. We need to be careful when we disable slither errors.

(Though disabling slither errors in here in the code is far better than disabling them in a separate json file, since this makes it easy to see them during review.)

Reply:
yes you are right slither was picking up the error for the right reasons. Manual disables removes here: https://github.com/OriginProtocol/origin-dollar/pull/2335/commits/08f6c83cb4e6c14b268ba1898a8ee20da5cbe81d